sources/packages/backend/apps/api/src/utilities/class-validation/custom-validators/is-number-greater-than.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+  registerDecorator,
+  ValidationOptions,
+} from "class-validator";
+
+/**
+ *  Check if the value is greater than the reference number.
+ */
+@ValidatorConstraint()
+class IsNumberGreaterThanConstraint implements ValidatorConstraintInterface {
+  validate(value: number, args: ValidationArguments): boolean {
+    const [condition] = args.constraints;
+    const referenceNumber = condition(args.object) as number;
+    if (referenceNumber === null || referenceNumber === undefined) {
```

Comment:
Is this validation missing the empty string?

Reply:
as the expected input is a number, didn't check the empty string. will it be useful to check?

Reply:
Is there anything that would prevent a number from reaching this code at runtime?
Is your point that this would probably be used in conjunction with the IsNumber? If yes, I do not see it as a blocker.